cmd/influx/v1_authorization.go

```diff
@@ -479,9 +519,78 @@ func v1AuthorizationInactiveF(cmd *cobra.Command, args []string) error {
 	})
 }
 
+var (
+	errMultipleMarchingAuthorizations = errors.New("multiple authorizations found")
+)
+
+func v1FindOneAuthorization(s *authorization.Client, filter influxdb.AuthorizationFilter) (*influxdb.Authorization, error) {
+	authorizations, _, err := s.FindAuthorizations(context.Background(), filter)
+	if err != nil {
+		if err.Error() == authorization.ErrAuthNotFound.Msg {
+			return nil, authorization.ErrAuthNotFound
+		}
+
+		return nil, err
+	}
+
+	if len(authorizations) > 1 {
+		return nil, errMultipleMarchingAuthorizations
+	}
+
+	return authorizations[0], nil
+}
+
+type v1AuthLookupFlags struct {
+	id       influxdb.ID
+	username string
+	required bool // required when set to true determines whether validate expects either id or username to be set
+}
+
+func (f *v1AuthLookupFlags) register(cmd *cobra.Command, persistent bool) {
+	opts := flagOpts{
+		{
+			DestP:      &f.id,
+			Flag:       "id",
+			Desc:       "The ID of the authorization",
+			Persistent: persistent,
+		},
+		{
+			DestP:      &f.username,
+			Flag:       "username",
+			Desc:       "The username of the authorization",
+			Persistent: persistent,
+		},
+	}
+	opts.mustRegister(cmd)
+}
+
+func (f *v1AuthLookupFlags) validate() error {
+	switch {
+	case f.id.Valid() && f.username != "":
+		return errors.New("specify id or username")
```

Comment:
Could we reword the error to make it more clear that it's an XOR? At first glance I wasn't sure why this was an error.

Reply:
How about 

> specify either id or username, not both

Reply:
LGTM